tests/v2/validation/nodemetadata/nodemetadata.go

```diff
@@ -0,0 +1,61 @@
+package nodemetadata
+
+import (
+	"fmt"
+	"net/url"
+
+	v1 "github.com/rancher/shepherd/clients/rancher/v1"
+	coreV1 "k8s.io/api/core/v1"
+)
+
+func getNodeSpec(nodeList *v1.SteveCollection) (*coreV1.Node, error) {
+	firstNode := nodeList.Data[0]
+	nodeSpec := &coreV1.Node{}
+	err := v1.ConvertToK8sType(firstNode, nodeSpec)
+	if err != nil {
+		return nil, err
+	}
+	return nodeSpec, nil
+}
+
+func verifyAnnotation(query url.Values, steveclient *v1.Client, present bool, key string, val string) (bool, *v1.SteveCollection, error) {
+	nodeList, err := steveclient.SteveType("node").List(query)
+	if err != nil {
+		return false, nodeList, err
+	}
+	nodeSpec, err := getNodeSpec(nodeList)
+	if err != nil {
+		return false, nodeList, err
+	}
+	value, exists := nodeSpec.ObjectMeta.Annotations[key]
+	if present {
+		if exists && value == val {
+			return true, nodeList, nil
+		}
+		return false, nodeList, fmt.Errorf("key %s does not exist with the value %s", key, value)
+	} else {
+		if !exists {
+			return true, nodeList, nil
+		}
+		return false, nodeList, fmt.Errorf("key %s should not be present", key)
+	}
+
+}
+
+func addAnnotation(nodeSpec *coreV1.Node, nodeList *v1.SteveCollection, steveclient *v1.Client, key string, value string) error {
+	nodeSpec.ObjectMeta.Annotations[key] = value
+	_, err := steveclient.SteveType("node").Update(&nodeList.Data[0], nodeSpec)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func deleteAnnotation(nodeSpec *coreV1.Node, nodeList *v1.SteveCollection, steveclient *v1.Client, key string) error {
```

Comment:
if you aren't using the whole SteveCollection in this function, I think it makes more sense to just send this the node you're interested in.

Reply:
Okay I see. But what is the difference if I use the whole `SteveCollection` vs just the Node's `Data []v1.SteveAPIObject`

Reply:
I see okay. But i would still say the function I have written is good.